framework/base/Action.php

```diff
@@ -115,8 +133,84 @@ protected function beforeRun()
     /**
      * This method is called right after `run()` is executed.
      * You may override this method to do post-processing work for the action run.
+     * @return mixed the processed action result got from `$this->result`.
      */
     protected function afterRun()
     {
     }
+
+    /**
+     * Resolves action arguments, derived class my override this method to provide custom argument resolver
+     *
+     * @return array
+     */
+    public function resolveActionArguments(array $params)
+    {
+        $args = $this->controller->bindActionParams($this, $params);
+        if (Yii::$app->requestedParams === null) {
+            Yii::$app->requestedParams = $args;
+        }
+        return $args;
+    }
+
+    /**
+     * Executes action handler using resolved action arguments
+     *
+     * @param array $args the action handler arguments
+     * @return mixed the result of action handler invocation
+     */
+    protected function executeAction($args)
+    {
+        if (!method_exists($this, 'run')) {
+            throw new InvalidConfigException(get_class($this) . ' must define a "run()" method.');
+        }
+        
+        if ($this->beforeRun()) {
+            Yii::debug('Running action: ' . get_class($this) . '::run(), invoked by '  . get_class($this->controller), __METHOD__);
+            $this->result = call_user_func_array([$this, 'run'], $args);
+            $this->afterRun();
+        }        
+    }
+    
+    /**
+     * Get argument passed to the specified parameter
+     *
+     * @param string $paramName name of declared parameter in inlineAction actionMethod/ Action::run().
+     * @return mixed value bound to the parameter `$paramName`
+     * @throws BadRequestHttpException if the parameter `$paramName` is not defined or argument not yet bound to it.
+     */
+    public function getRequestedParam($paramName)
+    { 
+        if (array_key_exists($paramName, (array) Yii::$app->requestedParams)) {
+
+            $found = false;           
+            if (!isset($this->controller->ActionInjectionsMeta[$paramName])) {
+                $requestedParam = Yii::$app->requestedParams[$paramName];
```

Comment:
I think immediate `return` here and in other cases below is better instead of if/else ladder.

Reply:
> 
> 
> I think immediate `return` here and in other cases below is better instead of if/else ladder.

Ah sorry @bizley  I have not yet read this comment. Anywhere I did not use this approach to not have so many `return ` s. However I made small changes in last comment to have more self explanatory and performant code. 

Thanks

Reply:
if/else ladder replaced by immediate returns as recommended

Thanks